lib/dpl/provider/pages.rb

```diff
@@ -45,9 +49,15 @@ def slug
       end
 
       def check_auth
+        unless @gh_token || @gh_deploy_key
```

Comment:
I've implemented this method for token version by talking to the api in #719.
You'll need to extend it with smth like `ssh -Tvvvvv git@github.com` to check whether the key is valid :)